pkgs/desktops/deepin/default.nix

```diff
@@ -12,6 +12,47 @@ let
           --replace "${x}" "${y}"
       done
     '';
+
+    getPatchFrom = commonRp:
+      let
+        rpstr = s1: s2: " --replace \"${s1}\" \"${s2}\"";
+        # rpstr "t1" "t2" => " --replace \"t1\" \"t2\""
+        rpstrL = l: if lib.length l == 2 then rpstr (lib.head l) (lib.last l) else (throw "input must be a list of 2 string: [ s1 s2 ]");
+        # rpstrL [ "t1" "t2" ] => rpstr "t1" "t2"
+        rpfile = filePath: replaceLists:
```

Comment:
Or just use an attrset instead of a list?

Reply:
Both should work, use list maybe a bit faster

Reply:
attrset should be more readable.

Reply:
\> a bit faster
\> nix

Not convincing.

Reply:
We should use the variant that is more readable here.

Reply:
Maybe we can avoid using this function